src/lib/scheduler/abstract_task.cpp

```diff
@@ -75,6 +75,17 @@ void AbstractTask::execute() {
   DebugAssert(!(_started.exchange(true)), "Possible bug: Trying to execute the same task twice");
   DebugAssert(is_ready(), "Task must not be executed before its dependencies are done");
 
+  // Besides being another sanity check, reading the atomic _is_scheduled is important for thread safety. We need to
```

Comment:
I don't understand the paragraph. Maybe this gets better when I look at the code in more detail and when I don't just give it a first scroll through.

Reply:
If not, let me know. As this is far from obvious, the comment should be understandable.